### PR TITLE
Add back support for multiple apps

### PR DESCRIPTION
**Pull request by rwwagner90**

### CODE REVIEW

**Comment by rwwagner90** [comment]
@rwjblue I was unable to get anything close to functioning here, but hopefully this shows you what I was going for. Perhaps you could help me figure out where I went wrong?

File: app/port.js [hunk 1]
```diff
@@ -3,39 +3,46 @@ import EmberObject, { computed } from '@ember/object';
...
 
-      if (!this.get('applicationId')) {
-        this.set('applicationId', message.applicationId);
+      if (!this.applicationId) {
+        this.set('applicationId', applicationId);
       }
 
       // save list of application ids
-      addIfNotPresent(detectedApplications, message.applicationId);
+      if (!this.detectedApplications..mapBy('applicationId').includes(applicationId)) {
```
Comment (rwjblue) [inline app/port.js]:
```suggestion
      if (!this.detectedApplications.mapBy('applicationId').includes(applicationId)) {
```

**Comment by alexhancock** [comment]
> but @teddyzeenny was saying the `uniqueId` stuff was necessary because `guidFor` has no guarantee of being unique across iframes.

That makes sense to me if we actually have code running across multiple execution contexts that `guidFor` wouldn't be guaranteed to generic unique IDs across them. I'm still learning how the whole architecture works.

Does the logic in `ember_debug/port.js` actually run in the context of multiple iframes if there are apps in iframes on the page, or is it all run centrally in one context?

**Comment by alexhancock** [comment]
@rwwagner90 Ahh, I see. Thanks for clarifying that!

**Comment by nummi** [comment]
Seems to be working on the Square Dashboard with an iframe inserted into the markup:

image

My test app has a blank `<option>` but it still functions:

image

**Comment by alexhancock** [comment]
@rwwagner90 Any interest in re-starting this? I think where we left off you wanted a code review from @teddyzeenny?

Seems like we had the solution pretty close in code. I would love to finish it if there is interest.

**Comment by rwwagner90** [comment]
@alexhancock I do plan to wrap this up soon. I spoke to @teddyzeenny and he is in favor of us shipping more code and worrying about it less. If it breaks things for anyone, hopefully they will report bugs. We currently have a lot of other issues to work through for inspector, and we have to work through those and get the tests passing again, before we can merge any PRs. Would you be interested in looking into some of the issues with us?

**Comment by alexhancock** [comment]
@rwwagner90 Sounds good. I am happy to spend some time looking into those issues. What's the best way for me to help? Is there a branch I should check out and try to debug things on my own, or should I join to discuss on discord at some time?

I am wrapping up my work day soon today, but should be able to help look into things tomorrow.

**Comment by rwwagner90** [comment]
@alexhancock I just rebased this with master. Would you mind testing to make sure it works for you?

**Comment by alexhancock** [comment]
@rwwagner90 sure thing - will test shortly

**Comment by alexhancock** [comment]
@rwwagner90 Seemed to work well for me. I tested our case where we have two ember apps on the page, and then I dropped an iframe in the page to a third ember app running on my computer, and they all showed up in the picker. When switching back and forth between the apps, I got this error on occasion (not every time):

Screen Shot 2019-04-17 at 2 56 04 PM

... but even though I got the error, everything continued to work.

**Comment by rwwagner90** [comment]
@alexhancock would you possibly be interested in debugging that error and trying to determine what is causing it?

**Comment by alexhancock** [comment]
@rwwagner90 sure - will look tomorrow.

**Comment by rwwagner90** [comment]
@alexhancock sounds good! Please keep me posted 😃

**Comment by rwwagner90** [comment]
@alexhancock any luck?

**Comment by alexhancock** [comment]
Got a little more clarity on what the error is, but didn't find a solution yet. Had limited time to look at it so far, but will try to figure it out today/tomorrow.

**Comment by rwwagner90** [comment]
@alexhancock just keep me posted! I would like to get this in soon, and release a new version 😃

**Comment by alexhancock** [comment]
@nummi @rwwagner90 I tested on our site (https://squareup.com/dashboard/). It works well from what I can see.

If we know of any other cases we should test them, but otherwise I suppose we just go for it?

**Comment by rwwagner90** [comment]
Merged. Thanks so much guys! If anything breaks, I'm sure people will let us know 😝

**Pull request merged**